### PR TITLE
47: IG detached signature waiver 007

### PR DESCRIPTION
**Pull request by christian-brindley**
Resolves: https://github.com/SecureBankingAccessToolkit/SecureBankingAcceleratorToolkit/issues/47

### CODE REVIEW

File: config/7.0/obdemo-bank/ig/scripts/groovy/JwkmsSignClientClaims.groovy [hunk 1]
```diff
@@ -27,24 +33,87 @@ if (!attributes.clientCertificate.privateKey) {
...
-                .kid(kid)
-                .done()
-                .claims(jwtClaims)
-                .build();
+if (signingOptions.unencodedpayload) {
+
+
+    OctetSequenceKey hmacJWK = OctetSequenceKey.parse("{"+
+            "\"kty\":\"oct\"," +
+            "\"k\":\"AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow\"" +
```
Comment (matt-wills) [inline config/7.0/obdemo-bank/ig/scripts/groovy/JwkmsSignClientClaims.groovy:66]: Is this temporary?

File: config/7.0/obdemo-bank/ig/scripts/groovy/ProcessDetachedSig.groovy [hunk 2]
```diff
@@ -2,11 +2,62 @@ import org.forgerock.http.protocol.*
...
+import groovy.json.JsonSlurper
+
+import com.nimbusds.jose.*;
+import com.nimbusds.jose.crypto.*;
+import com.nimbusds.jose.jwk.*;
+
+def validateUnencodedPayload(jws, payload) {
+    Payload detachedPayload = new Payload(payload);
+    JWSObject parsedJWSObject = JWSObject.parse(jws, detachedPayload);
+    JWKSet jwkSet = JWKSet.load(new URL(routeArgJwkUrl));
```
Comment (matt-wills) [inline config/7.0/obdemo-bank/ig/scripts/groovy/ProcessDetachedSig.groovy:14]: What happens if the JWKSet isn't loaded ok?

File: config/7.0/obdemo-bank/ig/scripts/groovy/JwkmsSignClientClaims.groovy (hunk 1 above)
Comment (christian-brindley) [reply config/7.0/obdemo-bank/ig/scripts/groovy/JwkmsSignClientClaims.groovy:66]: That's left over from initial version - have removed.

File: config/7.0/obdemo-bank/ig/scripts/groovy/ProcessDetachedSig.groovy (hunk 2 above)
Comment (christian-brindley) [reply config/7.0/obdemo-bank/ig/scripts/groovy/ProcessDetachedSig.groovy:14]: Have added exception handling for JWKS load

**Review by matt-wills — APPROVED** [review]

**Pull request merged**